### DISCUSSION TITLE
`base-url` replacer aliases vendors when directories monitored in the tsconfig `paths` have the same name

### DISCUSSION THREAD

**Discussion by valeriobelli** [justkey007/tsc-alias#148] [General]
Hi!

I'm unsure if I've found a bug or if it's misconfigured. Let's explain it with an example because it's peculiar.

Given this `tsconfig.json`

```json
{
  "compilerOptions": {
    "allowJs": false,
    "baseUrl": ".",
    "declaration": true,
    "declarationDir": "typings",
    "declarationMap": true,
    "esModuleInterop": true,
    "isolatedModules": false,
    "lib": ["esnext"],
    "module": "esnext",
    "moduleResolution": "node",
    "noEmit": true,
    "outDir": "lib",
    "paths": {
      "~/*": ["src/*"]
    },
    "skipLibCheck": true,
    "strict": true,
    "target": "es2021",
    "typeRoots": ["./types", "../../node_modules/@types"],
    "types": ["jest", "node"]
  },
  "include": ["src", "types"]
}
```

the following dependencies in the `package.json`

```json
{
  ...,
  "dependencies": {
    "@swc/helpers": ">=0.4.3 <1.0.0",
    "base64-js": "^1.5.1",
    "date-fns": "^2.28.0",
    "invariant": "^2.2.4",
  },
  "devDependencies": {
    "fp-ts": "^2.12.1",
    "io-ts": "^2.2.16",
    "react": "17.0.2",
  },
  "peerDependencies": {
    "fp-ts": "^2.12.1",
    "io-ts": "^2.2.16",
    "react": "17.0.2",
  },
  ...
}
```

the following package's folders
![image](https://user-images.githubusercontent.com/56547421/204760961-db164138-16b8-438d-8a37-eaa6ce36080b.png)

and the following file contained in `~/io-ts/applicationContext/index.ts`

```typescript
import { literal, TypeOf } from 'io-ts'

export const applicationContext = literal('native')

export type ApplicationContext = TypeOf<typeof applicationContext>
```

The replacement ends up with

```typescript
import { TypeOf } from '../../io-ts';
export declare const applicationContext: import("../../io-ts").LiteralC<"native">;
export declare type ApplicationContext = TypeOf<typeof applicationContext>;
```

As you can see, it seems like `tsc-alias` is treating vendors with homonyms folders as aliases. `io-ts` should have been resolved to the vendor, and not prefixed with the relative path to the `io-ts` folder. I suppose `base-url` is where this is happening. Is it due to a misconfiguration, or is it a bug?

Let me know if you need some more context.

**Comment by raouldeheer** [comment]
@valeriobelli this looks like a baseUrl issue. Please check the FAQ #110 question 1.

**Reply by valeriobelli** [reply]
I'm sorry for not getting back to you sooner. I wanted to try this change extensively due to the complex codebase my company works on.

Disabling the `baseUrl` replacer did the trick. Thanks!

**Comment by Dha0x** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。
